File: Api/Timesheet/Entities/EmployeeScheduleType.cs
namespace Timesheet.Entities
{
    public class EmployeeScheduleType : EntityBase
    {
        public DateTime InitialDate { get; set; }
        public DateTime EndDate { get; set; }

        public int ScheduleTypeId { get; set; }
        public ScheduleType? ScheduleType { get; set; }
        public  int EmployeeId { get; set; }
        public  Employee? Employee { get; set; }
    }
}

File: Web/src/app/models/records/employee-schedule-type.ts
import { Employee } from "./employee";
import { ScheduleType } from "./shedule-type";

export class EmployeeScheduleType {
    Id: number;
    ScheduleTypeId: number;
    EmployeeId: number;
    InitialDate: Date;
    EndDate: Date;

    ScheduleType?: ScheduleType;
    Employee?: Employee;
  
    constructor(params?: Partial<EmployeeScheduleType>) {
      this.Id = params?.Id;
      this.ScheduleTypeId = params?.ScheduleTypeId ;
      this.EmployeeId = params?.EmployeeId ;
      this.InitialDate = params?.InitialDate ;
      this.EndDate = params?.EndDate ;
      this.ScheduleType = params?.ScheduleType ;
      this.Employee = params?.Employee ;

    }
  }
  